crates/rolldown/src/module_finalizers/scope_hoisting/impl_visit_mut.rs

```diff
@@ -263,6 +264,24 @@ impl<'me, 'ast> VisitMut<'ast> for ScopeHoistingFinalizer<'me, 'ast> {
     walk_mut::walk_expression(self, expr);
   }
 
+  // foo.js `export const bar = { a: 0 }`
+  // main.js `import * as foo_exports from './foo.js';\n foo_exports.bar.a = 1;`
+  // The `foo_exports.bar.a` ast is `StaticMemberExpression(StaticMemberExpression)`, The outer StaticMemberExpression span is `foo_exports.bar.a`, the `visit_expression(Exprssion::MemberExpression)` is called with `foo_exports.bar`, the span is inner StaticMemberExpression.
```

Comment:
It is reason why the `visit_expression(Exprssion::MemberExpression)` not handle the case.

Reply:
If we migrated to  oxc `Traverse` API, the issue should be gone.